### PR TITLE
MPDX-9815 - Add completion state to sub-steps in questionnaire components

### PR DESCRIPTION
**Pull request by zweatshirt**
## Description

This adds a completion prop for SubStep array items based on which subsections of the form is completed. It is relatively simple since the form does not have many split-up sections.

https://jira.cru.org/browse/MPDX-9815

## Testing

- Go to NSO MPD Questionnaire
- Ideally, start with a fresh form
- Test the full flow and ensure that the substep circle icon fills in as the form is completed. Also good to double check here that the circular percentage bar works well.

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels (_Add the label "Preview" to automatically create a preview environment_)
- [ ] I have run the Claude Code `/pr-review` command locally and fixed any relevant suggestions
- [ ] I have requested a review from another person on the project
- [ ] I have tested my changes in preview or in staging
- [ ] I have cleaned up my commit history

### CODE REVIEW

**Comment by zweatshirt** [comment]
@canac Curious if I could get your opinion of the form's design in this preview! And if you had any constructive criticism over the UI/UX particularly. The form is ~95% done until stakeholders desire changes. After tying in the OneApp ministry list, it should be 100% unless I happened to miss anything.

File: src/components/HrTools/NsoMpdQuestionnaire/Shared/SubStepList.tsx [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
-import { ListItemContent } from 'src/components/HrTools/Shared/CalculationReports/StepsList/ListItemContent';
+import CircleIcon from '@mui/icons-material/Circle';
+import RadioButtonUncheckedIcon from '@mui/icons-material/RadioButtonUnchecked';
+import { List, ListItem, ListItemText, styled } from '@mui/material';
+import {
+  CategoryListItemIcon,
+  CategoryListItemStyles,
+} from 'src/components/HrTools/Shared/CalculationReports/Shared/styledComponents/StepsListStyles';
+
+const StyledCategoryListItemIcon = styled(CategoryListItemIcon, {
```
Comment (zweatshirt) [inline src/components/HrTools/NsoMpdQuestionnaire/Shared/SubStepList.tsx:10]: [Suggestion] This `StyledCategoryListItemIcon` re-implements the same two-state dot renderer (blue `CircleIcon` / gray `RadioButtonUnchecked`, `mpdxBlue.main` / `mpdxGrayDark.main`) that already exists in `GoalCalculator/SharedComponents/SectionList.tsx`. Consider extracting a shared 2-state dot renderer under `HrTools/Shared/` so the two flows stay in sync. Non-blocking. (The `styled()` + `shouldForwardProp` approach itself is fine.)

File: src/components/HrTools/NsoMpdQuestionnaire/PersonalInformation/PersonalInformation.tsx [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
 
   const subSteps: SubStep[] = [
-    { id: 'staff-information', title: t('Staff Information') },
-    { id: 'contact-information', title: t('Contact Information') },
+    // Staff Information is a read-only review card with nothing to fill in.
+    { id: 'staff-information', title: t('Staff Information'), complete: true },
+    {
+      id: 'contact-information',
+      title: t('Contact Information'),
+      complete: isStepComplete(
```
Comment (zweatshirt) [inline src/components/HrTools/NsoMpdQuestionnaire/PersonalInformation/PersonalInformation.tsx:23]: [Suggestion] The new per-page completion wiring (`isStepComplete(step, questionnaire)` → sub-step dot) has no integration test, and there is no `PersonalInformation.test.tsx`. A page passing the wrong step enum would still pass every existing test. Worth a page-level test asserting the Contact Information dot fills once `phoneNumber` (and, for married users, `spousePhoneNumber`) is present. The `SubStepList` presentational layer is already well covered.

File: src/components/HrTools/NsoMpdQuestionnaire/Shared/SubStepList.test.tsx [hunk 3]
```diff
@@ -34,4 +42,25 @@ describe('SubStepList', () => {
...
   });
+
+  it('leaves incomplete sub-step dots outlined even when current', () => {
+    const { getAllByTestId, queryByTestId } = render(<TestComponent />);
+
+    expect(getAllByTestId('RadioButtonUncheckedIcon')).toHaveLength(2);
+    expect(queryByTestId('CircleIcon')).not.toBeInTheDocument();
+  });
+
+  it('fills a sub-step dot blue once its data is complete', () => {
```
Comment (zweatshirt) [inline src/components/HrTools/NsoMpdQuestionnaire/Shared/SubStepList.test.tsx]: [Suggestion] Minor test robustness: this asserts one `CircleIcon` and one `RadioButtonUncheckedIcon` render, but not that the *complete* row is the filled one — a `complete ? outlined : filled` inversion could still pass. A row-scoped assertion (e.g. `within(getByText('1. Section One').closest('li')).getByTestId('CircleIcon')`) would catch it. Also consider an all-complete case (all filled, zero outlined).

File: src/components/HrTools/NsoMpdQuestionnaire/Shared/stepCompletion.ts [hunk 4]
```diff
@@ -75,10 +75,11 @@ const getRequiredFields = (
   return [...baseFields, ...variantFields];
 };
 
-const isStepComplete = (
+export const isStepComplete = (
```
Comment (zweatshirt) [inline src/components/HrTools/NsoMpdQuestionnaire/Shared/stepCompletion.ts:78]: [Suggestion] `isStepComplete` is now the exported entry point every page dot depends on and gained a new null guard. `stepCompletion.test.ts` exercises it only transitively via `getCompletionPercentage`. A direct unit test (`isStepComplete(step, null)` → `false`; per-step required-field cases including the married spouse-phone path) would pin the contract.

**Review by zweatshirt — COMMENTED** [review]
## 🤖 Multi-Agent Review — APPROVED WITH SUGGESTIONS

**Verdict:** ✅ APPROVED_WITH_SUGGESTIONS · **0 blockers** · Risk: MEDIUM

5 agents (Architecture, Testing, Standards, UX, Financial-domain). This is a presentational / state-plumbing change — no auth, API, GraphQL, schema, or dependency surface. The Financial domain agent confirmed there is no financial-calculation code here (`isStepComplete` only checks field *presence*).

**Resolved during review:** the unused `export` on `isFieldFilled` was made module-private.

**Non-blocking suggestions** (posted inline): extract the shared 2-state dot renderer, add page-level wiring tests, and a couple of minor test robustness / edge-case items.

**Considered and intentionally skipped:** adding `titleAccess` a11y labels to the dots. MUI's `titleAccess` renders an SVG `<title>`, which browsers surface as a hover tooltip — unwanted on these dots — and the completion state is decorative/redundant next to the already-readable sub-step label (and the gap pre-dates this PR). If screen-reader labelling is wanted later, visually-hidden text is the right mechanism, not `titleAccess`.
